foundation/foundation-archetype/src/main/resources/archetype-resources/deploy.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+## ArgoCD local deployment script
+if [ "$1" == "up" ]; then
+  argocd app get ${artifactId} --server localhost:30080 --plaintext &> /dev/null
+
+  if [ $? -eq 0 ]; then
+    echo "${artifactId} is deployed"
+  else
+    argocd app create ${artifactId} \
+      --server localhost:30080 --plaintext \
+      --dest-namespace ${artifactId} \
+      --dest-server https://kubernetes.default.svc \
+      --repo ${projectGitUrl} \
+      --path ${artifactId}-deploy/src/main/resources \
+      --revision main \
+      --values values.yaml \
+      --values values-dev.yaml \
+      --sync-policy automated
+  fi
+else
+  if [ "$1" == "down" ]; then
+    argocd app delete ${artifactId} --server localhost:30080 --plaintext
+  else
+    echo "error: you must use \"up\" or \"down\" command for \"deploy\""
+  fi
+fi
```

Comment:
S: print usage if neither of the other branches hit. Also would pull them out into functions to make it a little more readable, e.g. `deploy`, `delete`, and `print_usage`